mm2src/mm2_bitcoin/chain/src/block_header.rs

```diff
@@ -242,7 +241,7 @@ impl Deserializable for BlockHeader {
         };
         let nonce = if is_zcash {
             BlockHeaderNonce::H256(reader.read()?)
-        } else if (version == KAWPOW_VERSION && !reader.coin_variant().is_btc())
+        } else if (version == KAWPOW_VERSION && reader.coin_variant().is_rvn())
             || version == MTP_POW_VERSION && time >= PROG_POW_SWITCH_TIME
```

Comment:
`&&` gets computed before `||` (https://doc.rust-lang.org/reference/expressions.html#r-expr.precedence)
let's put some parentheses around
`version == MTP_POW_VERSION && time >= PROG_POW_SWITCH_TIME`
to be consistent or remove the parentheses from
`(version == KAWPOW_VERSION && reader.coin_variant().is_rvn())`